### PR TITLE
fix: Runtime dependencies and some cleanup

### PR DESCRIPTION
**Pull request by LecrisUT**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by LecrisUT** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -40,6 +40,8 @@ requirements:
   run:
     - python
     - {{ pin_compatible('numpy') }}
+    - typing-extensions    # [py<39]
```
[the remark this answers is not in the document]
Comment (LecrisUT) [reply recipe/meta.yaml:44]: `conda-build .`? :shrug:

[the remark this answers is not in the document]
Comment (atztogo) [reply recipe/meta.yaml:44]: For example, if I conda-build for phonopy, this downloads currently available spglib. But I don't know how to download the package corresponding to this PR.

[the remark this answers is not in the document]
Comment (LecrisUT) [reply recipe/meta.yaml:44]:
```console
$ git clone https://github.com/conda-forge/spglib-feedstock
$ cd spglib-feedstock
$ git remote add LecrisUT https://github.com/LecrisUT/spglib-feedstock/
$ git fetch LecrisUT
$ git checkout LecrisUT/fix/dependencies
```
Alternatively you can try [this one](https://stackoverflow.com/a/30584951/22352077) which is more generic
```
$ git fetch origin pull/113/head:pr_113
$ git checkout pr_113
```

[the remark this answers is not in the document]
Comment (atztogo) [reply recipe/meta.yaml:44]: I follow your first commands of git, and run `conda build .`, but I failed.

```
Could not solve for environment specs
The following package could not be installed
└─ c_osx-arm64 does not exist (perhaps a typo or a missing channel).
```

I can not understand what you want me to do.

[the remark this answers is not in the document]
Comment (LecrisUT) [reply recipe/meta.yaml:44]: This seems like conda bug. Try to update `conda-build`/`conda` to the latest version. The rendered version seems to have used `conda-build 24.3.0`, which version do you have?

[the remark this answers is not in the document]
Comment (atztogo) [reply recipe/meta.yaml:44]:
```
conda                     24.3.0           py38h10201cd_0    conda-forge
conda-build               24.3.0           py38h10201cd_1    conda-forge
```

[the remark this answers is not in the document]
Comment (LecrisUT) [reply recipe/meta.yaml:44]: Hmm, let's see if people on matrix have some comments ([link](https://matrix.to/#/!SOyumkgPRWoXfQYIFH:matrix.org/$171500656982367WGWMN:matrix.org?via=matrix.org&via=gitter.im&via=robbinsa.me))

[the remark this answers is not in the document]
Comment (atztogo) [reply recipe/meta.yaml:44]: This suggestion went succeeded.

File: recipe/COPYING [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2014, Atsushi Togo
+Copyright (c) 2024, Spglib team
```
Comment (bollwyvl) [inline recipe/COPYING:1]: this file is in the upstream tarball, and can be removed

**Pull request merged**